From d0243a9f9961f0afdc09b555821b88edb2488be9 Mon Sep 17 00:00:00 2001
From: Stas Boukarev <stassats@gmail.com>
Date: Wed, 30 Sep 2020 02:46:31 +0300
Subject: [PATCH] Fix CPUID checking.

cpu-has- is an index, not a mask. Have popcnt and no avx => bad time.
---
 src/compiler/x86-64/parms.lisp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/compiler/x86-64/parms.lisp b/src/compiler/x86-64/parms.lisp
index 5aeddff47c..a9d2bef01e 100644
--- src/compiler/x86-64/parms.lisp
+++ src/compiler/x86-64/parms.lisp
@@ -178,8 +178,8 @@
 (defvar *binding-stack-pointer*)
 
 ;;; Bit indices into *CPU-FEATURE-BITS*
-(defconstant cpu-has-ymm-registers   1)
-(defconstant cpu-has-popcnt          2)
+(defconstant cpu-has-ymm-registers   0)
+(defconstant cpu-has-popcnt          1)
 
 (defconstant-eqx +static-symbols+
  `#(,@+common-static-symbols+
